### ISSUE TITLE
Header-like structures in content should be escaped

### ISSUE THREAD

**Issue by marijnvdwerf** [thephpleague/html-to-markdown#76]
**Input**:
```html
<p>Foo<br>--<br>Bar</p>
<p>Foo<br>Bar<br>--</p>
```

**Actual**:
```md
Foo  
--  
Bar

Foo  
Bar  
--
```

**Expected**:
```md
Foo  
\--  
Bar

Foo  
Bar  
\--
```

**Comment by andreskrey** [comment]
I don't understand why the last `--` is not escaped. What's the logic behind this? Only the `--` tags should be escaped if they have more text below?

**Comment by marijnvdwerf** [comment]
This was changed in version 0.24 of the spec. 'setext' headings couldn't span multiple lines before. Updated the expected output, thanks!

**Comment by andreskrey** [comment]
I could fix this in a similar way I did for #77, but since there are other escaping issues waiting to be fixed, adding new rules to that same chunk of code will make it quite messy.

I'm thinking of creating a separate function, like `escapeSpecialCharacters`, that will call other functions (like `escapeBlockquotelikeCharacters`, `escapeHeaderlikeCharacters`, etc) but I'm not sure what's the correct approach.

Since I'll be adding new functions to the ParagraphConverter class, should I also create a proper interface for it that would define this escaping functions? Is this necessary? Is there a better approach for this issue? Perhaps instead of escaping this characters at the Converter level, another solution could be to create a function inside the HtmlConverter Class, within the convertChildren function and before the convertToMarkdown call, to catch the string before the convertion and sanitize it there.

Any thoughts @colinodell?

**Comment by andreskrey** [comment]
I just went ahead and created a PR for this. It's #105

**Comment by colinodell** [comment]
Fixed via #105
